apps/api/src/app/middleware/apilogger.middleware.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable, Logger, NestMiddleware } from '@nestjs/common'
+import { Request, Response, NextFunction } from 'express'
+
+@Injectable()
+class ApiLoggerMiddleware implements NestMiddleware {
+  private readonly logger = new Logger('HTTP')
+
+  use(request: Request, response: Response, next: NextFunction) {
+    response.on('finish', () => {
+      const { method, originalUrl } = request
+      const { statusCode, statusMessage } = response
+
+      const message = `${method} ${originalUrl} ${statusCode} ${statusMessage}`
+
+      if (statusCode >= 500) {
+        return this.logger.error(message)
+      }
+
+      if (statusCode >= 400) {
+        return this.logger.warn(message)
+      }
+
+      return this.logger.log(message)
```

Comment:
very useful addition of the logging middleware!
let's just not print every request but only the problematic ones - may be just comment out the logger.log(message) for those who want it, but no need on production.

Reply:
changed to debug level, so it does not show on prod